doc/manual/packages.rst

```diff
@@ -346,6 +346,223 @@ Once you do this, the package manager knows your GitHub user name and can config
+to make changes.  While there are several possible approaches, here is
+one that is widely used:
+
+- From within Julia, type ``Pkg.checkout("Foo")``. This ensures you're
+  running the latest code (the ``master`` branch), rather than just
+  whatever "official release" version you have installed. (If you're
+  planning to fix a bug, at this point it's a good idea to check again
+  whether the bug has already been fixed by someone else. If it has,
+  you can request that a new official release be tagged so that the
+  fix gets distributed to the rest of the community.) If you receive
+  an error ``Foo is dirty, bailing``, see :ref:`Dirty packages
+  <man-pkg-dirty>` below.
+
+- Create a branch for your changes: from the command line (not within
+  Julia), navigate to the package folder (e.g.,
+  ``~/.julia/v0.4/Foo/``) and create a new branch using ``git
+  checkout -b <newbranch>``, where ``<newbranch>`` might be some
+  descriptive name (e.g., ``fixbar``). By creating a branch, you
+  ensure that you can easily go back and forth between your new work
+  and the current ``master`` branch (see
+  `<http://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell>`_).
+
+  If you forget to do this step until after you've already made some
+  changes, don't worry: see :ref:`more detail about branching
+  <man-post-hoc-branching>` below.
+
+- Make your changes. Whether it's fixing a bug or adding new
+  functionality, in most cases your change should include updates to
+  both the ``src/`` and ``test/`` folders.  If you're fixing a bug,
+  add your minimal example demonstrating the bug (on the current code)
+  to the test suite; by contributing a test for the bug, you ensure
+  that the bug won't accidentally reappear at some later time due to
+  other changes.  If you're adding new functionality, creating tests
+  demonstrates to the package owner that you've made sure your code
+  works as intended.
+
+- Run the package's tests and make sure they pass.  From
+  julia, you can do this in either of the following ways:
+
+  + ``Pkg.test("Foo")``: this will run your tests in a separate (new)
+    julia process.
+  + ``include("runtests.jl")`` from the package's ``test/`` folder (it's
+    possible the file has a different name, look for one that runs all
+    the tests): this allows you to run the tests repeatedly in the
+    same session without reloading all the package code; for packages
+    that take a while to load, this can be much faster. With this
+    approach, you do have to do some extra work to make :ref:`changes
+    in the package code <man-workflow-tips>`.
+
+  Alternatively, run the tests from the command line with ``julia
+  runtests.jl`` from within the package's ``test/`` folder.
+
+
+- Commit your changes: see `<http://git-scm.com/book/en/v2/Git-Basics-Recording-Changes-to-the-Repository>`_.
+
+- Submit your changes: from within julia, type
+  ``Pkg.submit("Foo")``. This will push your changes to your GitHub
```

Comment:
That's described in the section immediately above this one. But I can insert a reference.